### PR TITLE
products rpc methods

### PR DESCRIPTION
**Pull request by furkhat**

### CODE REVIEW

File: ui/generic.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package ui
```
Comment (dzeckelev) [inline ui/generic.go]: We agreed not to transfer this code.

**Comment by furkhat** [comment]
@dzeckelev updated.

File: doc/ui/rpc.md [hunk 2]
```diff
@@ -33,6 +33,45 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
+#### Get Products
+
+*Method*: `getProducts`
+
+*Description*: Get all products available to the agent.
+
+*Parameters*: None.
+
+*Result (object)*:
+1. Products (Array of data.Product)
```
Comment (ababo) [inline doc/ui/rpc.md]: 1. Products (array of `data.Product` objects)

File: ui/misc.go [hunk 3]
```diff
@@ -42,6 +42,31 @@ func (h *Handler) findByColumn(logger log.Logger, notFoundError error,
...
+
+func (h *Handler) update(logger log.Logger, rec reform.Record) error {
+	if err := h.db.Update(rec); err != nil {
+		logger.Error(err.Error())
+		return ErrInternal
+	}
+	return nil
+}
+
+func (h *Handler) operateOnRecord(logger log.Logger, rec reform.Record,
```
Comment (ababo) [inline ui/misc.go]: Do we really need that function?

**Comment by furkhat** [comment]
@simon @dzeckelev updated.

File: doc/ui/rpc.md [hunk 4]
```diff
@@ -31,7 +31,46 @@ This document describes a UI JSON RPC API located in "ui" namespace.
 2. Channel id (string)
 3. Gas price (number)
 
-*Result (object)*: Empty
+*Result (object)*: None.
```
Comment (ababo) [inline doc/ui/rpc.md]: Please remove `(object)`.

**Comment by furkhat** [comment]
@simon updated.

**Pull request merged**